src/code/z_room.c

```diff
@@ -354,51 +388,50 @@ void Room_Draw1Single(PlayState* play, Room* room, u32 flags) {
         gSPSegment(POLY_XLU_DISP++, 0x03, room->segment);
         Gfx_SetupDL_25Xlu(play->state.gfxCtx);
         gSPMatrix(POLY_XLU_DISP++, &gMtxClear, G_MTX_MODELVIEW | G_MTX_LOAD);
-        gSPDisplayList(POLY_XLU_DISP++, meshHeader1Entry->xlu);
+        gSPDisplayList(POLY_XLU_DISP++, dListsEntry->xlu);
     }
 
     CLOSE_DISPS(play->state.gfxCtx, "../z_room.c", 691);
 }
 
-BgImage* func_80096A74(MeshHeader1Multi* meshHeader1Multi, PlayState* play) {
+BgImage* Room_GetMultiBackgroundImage(MeshHeaderPrerenderMulti* meshHeaderPrerenderMulti, PlayState* play) {
     Camera* activeCam = GET_ACTIVE_CAM(play);
-    s32 camDataIdx;
-    s16 camDataIdx2;
+    s32 index;
+    s16 newIndex;
     Player* player;
     BgImage* bgImage;
     s32 i;
 
-    camDataIdx = activeCam->camDataIdx;
-    // jfifid
-    camDataIdx2 = func_80041C10(&play->colCtx, camDataIdx, BGCHECK_SCENE)[2].y;
-    if (camDataIdx2 >= 0) {
-        camDataIdx = camDataIdx2;
+    index = activeCam->camDataIdx;
+    newIndex = func_80041C10(&play->colCtx, index, BGCHECK_SCENE)[2].y;
+    if (newIndex >= 0) {
+        index = newIndex;
     }
```

Comment:
So could you explain to me what `index`/`newIndex` is being used for in this context? This is all `BgCamInfo` code, with `func_80041C10` getting the `bgCamFuncData`, and in this particular case, `[2].y` is getting what's called `jfifId`, is that accurate given the docs here? So `index` is a `bgCamDataIndex` but `newIndex` is a `jfjfId` but they're being compared and overwritten with each other. So is `jfifId` correct?

Reply:
@engineer124 Can I get your thoughts on the latest iteration of this that I did I forgot when?